Updated docs and added deprecation warnings to acknowledge a bool tensor

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs and added deprecation warnings to acknowledge a bool tensor**
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Updated docs to acknowledge a bool tensor

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

gchanan commented on torch/_tensor_docs.py:
```diff
@@ -362,15 +362,13 @@ def add_docstr_all(method, docstr):
 Example::
 
-    >>> a = torch.randn(4, 2).byte() % 2
+    >>> a = torch.rand(1, 2).bool()
```
these examples seem bogus -- you aren't really ever going to get False, right?

How about leaving these as randn and using a comparison operator (like `< 0`) instead of converting directly to bool?

gchanan reviewed (approved):
looks good if you change the examples.

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Updated docs to acknowledge a bool tensor

gh-metadata: pytorch pytorch 22261 gh/izdeby/11/head

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs to acknowledge a bool tensor**
* #21113 [BC-BREAKING] [WIP] Changed tensor comparison return type from uint8 to bool
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs to acknowledge a bool tensor**
* #21113 [BC-BREAKING] [WIP] Changed tensor comparison return type from uint8 to bool
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs to acknowledge a bool tensor**
* #21113 [BC-BREAKING] [WIP] Changed tensor comparison return type from uint8 to bool
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs to acknowledge a bool tensor**
* #21113 [BC-BREAKING] [WIP] Changed tensor comparison return type from uint8 to bool
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22261 Updated docs to acknowledge a bool tensor**
* #21113 [BC-BREAKING] [WIP] Changed tensor comparison return type from uint8 to bool
-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)

izdeby commented:
@gchanan, @ezyang  can you, please, take a final quick look before i land? 
Lint will be [fixed](https://github.com/pytorch/pytorch/pull/23736) here, to keep this change clean.

izdeby committed:
Update on "Updated docs and added deprecation warnings to acknowledge a bool tensor"

-------------
Change: 
- Updated docs to mention bool tensor functionality.
- Added deprecation warnings to masked methods.

Differential Revision: [D16005990](https://our.internmc.facebook.com/intern/diff/D16005990)
